src/analysis/normed_space/hahn_banach.lean

```diff
@@ -45,6 +46,60 @@ end
 
 end basic
 
+section complex
+variables {F : Type*} [normed_group F] [normed_space ℂ F]
+
+/-- Restrict a ℂ-subspace to an ℝ-subspace. -/
+noncomputable def restrict_scalars (p: subspace ℂ F) : subspace ℝ F := p.restrict_scalars ℝ ℂ F
+
+private lemma apply_real (p : subspace ℂ F) (f' : p →L[ℝ] ℝ) :
+  ∃ g : F →L[ℝ] ℝ, (∀ x : restrict_scalars p, g x = f' x) ∧ ∥g∥ = ∥f'∥ :=
+  exists_extension_norm_eq (p.restrict_scalars ℝ ℂ F) f'
```

Comment:
Inlining this into the proof below leads to errors like

```
type mismatch at application
  exists_extension_norm_eq (subspace.restrict_scalars ℝ ℂ F p)
term
  subspace.restrict_scalars ℝ ℂ F p
has type
  subspace ℝ (module.restrict_scalars ℝ ℂ F) : Type u_1
but is expected to have type
  subspace ℝ ?m_1 : Type ?
```

Reply:
Maybe add `--` comment explaining why you're making this definition?